packages/client/ui/react-ui/src/hooks/useRefreshToken.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useEffect, useRef } from "react";
+
+import { CrossmintAuthService, getJWTExpiration } from "@crossmint/client-sdk-auth-core";
+
+import { REFRESH_TOKEN_PREFIX, getCookie } from "../utils/authCookies";
+
+// 2 minutes before jwt expiration
+const TIME_BEFORE_EXPIRING_JWT_IN_SECONDS = 120;
+
+export type AuthMaterial = {
+    jwtToken: string;
+    refreshToken: {
+        secret: string;
+        expiresAt: string;
+    };
+};
+
+type UseAuthTokenRefreshProps = {
+    crossmintAuthService: CrossmintAuthService;
+    setAuthMaterial: (authMaterial: AuthMaterial) => void;
+};
+
+export function useRefreshToken({ crossmintAuthService, setAuthMaterial }: UseAuthTokenRefreshProps) {
+    const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+    const refreshAuthMaterial = useCallback(() => {
+        const refreshToken = getCookie(REFRESH_TOKEN_PREFIX);
+        if (refreshToken != null) {
+            crossmintAuthService
+                .refreshAuthMaterial(refreshToken)
+                .then((result) => {
```

Comment:
I'm curious about using then catch instead of async await. I prefer us using async/await to keep consistency with the codebase, although there may be a reason behind using it with this way

Reply:
My bad, I was doing it like that in a `useEffect` and forgot to change to async/await upon refactoring